js/src/sdk/models/Entity.ts

```diff
@@ -9,11 +9,34 @@ import { Triggers } from "./triggers";
 import { CEG } from "../utils/error";
 import logger from "../../utils/logger";
 import { SDK_ERROR_CODES } from "../utils/errors/src/constants";
-
+import { z } from "zod";
 const LABELS = {
   PRIMARY: "primary",
 };
 
+const ZExecuteActionParams = z.object({
```

Comment:
The Zod schema validation is a good addition for type safety, but consider adding more descriptive error messages for validation failures. For example:

```typescript
const ZExecuteActionParams = z.object({
  actionName: z.string({
    required_error: "Action name is required",
    invalid_type_error: "Action name must be a string"
  }),
  params: z.record(z.any()).optional(),
  text: z.string().optional(),
  connectedAccountId: z.string().optional()
});
```

This will help developers understand validation failures better.